app/views/documents/history.html.erb

```diff
@@ -15,61 +15,15 @@
           <% displayed_edition_number = entry.edition.number %>
         <% end %>
 
-        <% if entry.imported_from_whitehall? %>
+        <% if entry.whitehall_migration? && entry.details.imported_from_whitehall? %>
           <%= render "govuk_publishing_components/components/notice", {
-            title: "#{t "documents.history.entry_types.#{entry.entry_type}"}",
+            title: "#{t "documents.history.entry_types.whitehall_migration.#{entry.details.entry_type}"}",
             description_text: format_date(entry.created_at)
           } %>
+        <% elsif entry.whitehall_migration? %>
+          <%= render partial: "documents/whitehall_imported_entry", locals: {entry: entry} %>
         <% else %>
-
-          <h4 class="govuk-heading-s govuk-!-margin-bottom-1">
-            <%= t "documents.history.entry_types.#{entry.entry_type}" %>
-          </h4>
-
-          <div class="app-timeline-entry__dateline">
-            <%= format_date_at_time(entry.created_at) %>
-            <% if entry.created_by %>
-              by <%= name_or_fallback(entry.created_by) %>
-            <% end %>
-          </div>
-
-          <% if entry.internal_note? && entry.details %>
-            <%= simple_format(escape_and_link(entry.details.body)) %>
-          <% end %>
-
-          <% if (entry.withdrawn? || entry.withdrawn_updated?) && entry.details %>
-            <%= entry.details.public_explanation %>
-          <% end %>
-
-          <% if entry.backdated? %>
-            <% date = format_date(entry.revision.backdated_to) %>
-            <%= t "documents.history.entry_content.backdated", date: date %>
-          <% end %>
-
-          <% if entry.removed? && entry.details %>
-            <% removal = entry.details %>
-            <% if removal.explanatory_note.present? %>
-              <p><%= removal.explanatory_note %></p>
-            <% end %>
-
-            <% if removal.alternative_path.present? %>
-              <% if removal.redirect? %>
-                <p>
-                  Redirected to
-                  <%= link_to(Plek.new.website_root + removal.alternative_path,
-                              nil,
-                              class: "govuk-link") %>
-                </p>
-              <% else %>
-                <p>
-                  Alternative path
-                  <%= link_to(Plek.new.website_root + removal.alternative_path,
-                              nil,
-                              class: "govuk-link") %>
-                </p>
-              <% end %>
-            <% end %>
-          <% end %>
+          <%= render partial: "documents/timeline_entry", locals: {entry: entry} %>
```

Comment:
I think we hit a bit of a naming conundrum here where this partial name would imply that this can render all timeline entries and not just non-Whitehall ones. I think we should try resolve this before merging so we don't risk it festering as an area of communication.

Approaches we could use are:

1. Keep non-Whitehall timeline entries in this view file and only use a partial for them
2. Render everything in the _timeline_entry partial and inside that have a conditional that checks for whitehall_migration
3. Something I've not thought of

of those I find 2 most appealing and would work with the approach you've taken for tests.

Reply:
I went with option 2...